packages/cfd/src/Containers/jurisdiction-modal/jurisdiction-modal.tsx

```diff
@@ -0,0 +1,248 @@
+import React from 'react';
+import { Button, Modal, DesktopWrapper, MobileDialog, MobileWrapper, UILoader } from '@deriv/components';
+import { localize, Localize } from '@deriv/translations';
+import { connect } from 'Stores/connect';
+import RootStore from 'Stores/index';
+import { GetAccountSettingsResponse, GetSettings } from '@deriv/api-types';
+import JurisdictionModalContent from './jurisdiction-modal-content';
+import { WS, getAuthenticationStatusInfo, isMobile } from '@deriv/shared';
+import { TJurisdictionModalProps } from '../props.types';
+
+const JurisdictionModal = ({
+    account_status,
+    account_settings,
+    account_type,
+    disableApp,
+    enableApp,
+    is_jurisdiction_modal_visible,
+    is_eu,
+    is_virtual,
+    jurisdiction_selected_shortcode,
+    openPasswordModal,
+    real_synthetic_accounts_existing_data,
+    real_financial_accounts_existing_data,
+    trading_platform_available_accounts,
+    toggleCFDPersonalDetailsModal,
+    toggleJurisdictionModal,
+    setAccountSettings,
+    setJurisdictionSelectedShortcode,
+    toggleCFDVerificationModal,
+}: TJurisdictionModalProps) => {
+    const [checked, setChecked] = React.useState(false);
+    const [has_submitted_personal_details, setHasSubmittedPersonalDetails] = React.useState(false);
+
+    const {
+        poi_pending_for_vanuatu,
+        poi_verified_for_vanuatu,
+        poi_pending_for_bvi_labuan_maltainvest,
+        poi_resubmit_for_bvi_labuan_maltainvest,
+        poi_resubmit_for_vanuatu,
+        poi_verified_for_bvi_labuan_maltainvest,
+        poi_or_poa_not_submitted,
+    } = getAuthenticationStatusInfo(account_status);
+
+    React.useEffect(() => {
+        if (is_jurisdiction_modal_visible) {
+            setJurisdictionSelectedShortcode('');
+            if (!has_submitted_personal_details) {
+                let get_settings_response: GetSettings = {};
+                if (!account_settings) {
+                    WS.authorized.storage.getSettings().then((response: GetAccountSettingsResponse) => {
+                        get_settings_response = response.get_settings as GetSettings;
+                        setAccountSettings(response.get_settings as GetSettings);
+                    });
+                } else {
+                    get_settings_response = account_settings;
+                }
+                const { citizen, place_of_birth, tax_residence, tax_identification_number, account_opening_reason } =
+                    get_settings_response as GetSettings;
```

Comment:
```suggestion
                    get_settings_response;
```
is this cast necessary?